tools/serverpod_cli/lib/src/analyzer/entities/validation/restrictions.dart

```diff
@@ -139,7 +139,19 @@ class Restrictions {
     dynamic content,
     SourceSpan? span,
   ) {
-    if (content != null && !StringValidators.isValidTableIndexName(content)) {
+    if (content == null) return [];
+
+    if (documentDefinition is ClassDefinition &&
+        (documentDefinition as ClassDefinition).tableName == null) {
```

Comment:
I think

```dart
if (documentDefinition is ClassDefinition &&
        (documentDefinition as ClassDefinition).tableName == null)
```

can be written as

```dart
documentDefinition is ClassDefinition &&
        documentDefinition.tableName == null
```

But possibly not if `documentDefinition` is a non-local variable.